File: static_cdn/media_root/submissions/stonedcoder/SAFE.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
	int t,n,a,count;
	long int b,l,r;
	cin>>t;
	while(t--)
	{
		cin>>n>>a>>b;
		count=0;
		long int arr[n];
		 l=1-b;
		 r=1+b;
		for(int i=0;i<n;++i)
		{
			cin>>arr[i];
		}
		for(int i=0;i<n;++i)
		{
			if(arr[i]<=r and arr[i]>=l)
			{
				count++;
			}
		}
		if(count>=a)
		{
			cout<<"N0"<<'\n';
		}
		else
		{
			cout<<"YES"<<'\n';
		}
//		cout<<count;
	}
	return 0;
}

File: static_cdn/media_root/submissions/adiitrack/SAFE.py
t2=int(input())
t3,t4,t5=input().split( )
t3=int(t3)
t4=int(t4)
t5=int(t5)
arr=[]
arr=input().split( )
for i in range(len(arr)):
    arr[i]=int(arr[i])
count=0
for i in range(0,t3):
    if arr[i]<=1+t5 & arr[i]>=1-t5:
        count+=1 
    else:
        print("NO")
        break
if count==t3:
    print("YES")
    
    
